mock/py/mockbuild/file_downloader.py

```diff
@@ -42,28 +42,29 @@
 
         cls._initialize()
         try:
-            url = pkg
-            log.info('Fetching remote file %s', url)
-            req = requests.get(url)
-            req.raise_for_status()
-            if req.status_code != requests.codes.ok:
-                log.error("Can't download {}".format(url))
-                return False
+            log.info('Fetching remote file %s', pkg)
+            return cls._get_inner(pkg)
+        except requests.exceptions.RequestException as e:
+            log.error('Downloading error %s: %s', pkg, str(e))
+            return None
 
-            filename = urlsplit(req.url).path.rsplit('/', 1)[1]
-            if 'content-disposition' in req.headers:
-                _, params = cgi.parse_header(req.headers['content-disposition'])
-                if 'filename' in params and params['filename']:
-                    filename = params['filename']
-            pkg = cls.tmpdir + '/' + filename
-            with open(pkg, 'wb') as filed:
-                for chunk in req.iter_content(4096):
-                    filed.write(chunk)
-            cls.backmap[pkg] = url
-            return pkg
-        except Exception as err:  # pylint: disable=broad-except
-            log.error('Downloading error %s: %s', url, str(err))
-        return None
+    @classmethod
+    @backoff.on_exception(backoff.expo, requests.exceptions.RequestException, max_tries=3, max_time=10)
+    def _get_inner(cls, url):
+        req = requests.get(url)
+        req.raise_for_status()
+
+        filename = urlsplit(req.url).path.rsplit('/', 1)[1]
+        if 'content-disposition' in req.headers:
+            _, params = cgi.parse_header(req.headers['content-disposition'])
+            if 'filename' in params and params['filename']:
+                filename = params['filename']
+        pkg = cls.tmpdir + '/' + filename
+        with open(pkg, 'wb') as filed:
+            for chunk in req.iter_content(4096):
+                filed.write(chunk)
+        cls.backmap[pkg] = url
```

Comment:
This is not a new thing, but reported because it previously was in `get()`, now itis in `_get_inner()` (it appears to be a new warning - but it isn't).